### PR TITLE
Add test cases for volume set the snapshot data integrity and fix a typo

### PR DESCRIPTION
**Pull request by roger-ryao**
issue https://github.com/longhorn/longhorn/issues/4845

Signed-off-by: Roger Yao <roger.yao@suse.com>

### CODE REVIEW

**Comment by roger-ryao** [comment]
the `test_snapshot.py` passed, but the regression failed yesterday. I need to check the failed report.

https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2850/
Screenshot_20230106_091743

https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2851/
Screenshot_20230106_091326

File: manager/integration/tests/test_snapshot.py [hunk 1]
```diff
@@ -172,7 +204,7 @@ def test_snapshot_hash_global_enabled_and_per_volume_disable_and_with_immediate_
                                   SNAPSHOT_DATA_INTEGRITY_DISABLED)
 
 
-def test_snapshot_hash_global_enabled_and_per_volume_enable_mode_without_immediate_hash(client, volume_name, settings_reset):  # NOQA
+def test_snapshot_hash_global_enabled_and_per_volume_disable_and_without_immediate_hash(client, volume_name, settings_reset):  # NOQA
```
Comment (roger-ryao) [inline manager/integration/tests/test_snapshot.py:207]: Hi @derekbit 
I think this is a typo, so I modify it.

File: manager/integration/tests/test_snapshot.py [hunk 2]
```diff
@@ -220,7 +284,7 @@ def prepare_settings_for_snapshot_test(client, data_integrity, immediate_check,
     # Make the next hash time more predictable
     now = datetime.datetime.now()
     minute = (now.minute + period_in_minute) % 60
-    hour = now.hour + (now.minute + period_in_minute) / 60
+    hour = (now.hour + (now.minute + period_in_minute) / 60) % 24
```
Comment (roger-ryao) [inline manager/integration/tests/test_snapshot.py:287]: Fix the [failed case ](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2851/testReport/junit/tests/test_snapshot/test_snapshot_hash_global_enabled_and_per_volume_disable_and_without_immediate_hash/)

File: manager/integration/tests/test_snapshot.py [hunk 3]
```diff
@@ -187,6 +219,37 @@ def test_snapshot_hash_global_enabled_and_per_volume_enable_mode_without_immedia
...
+    """
+    prepare_settings_for_snapshot_test(client,
+                                       SNAPSHOT_DATA_INTEGRITY_FAST_CHECK,
+                                       "false",
+                                       "true")
+    check_per_volume_hash_disable(client, volume_name,
+                                  SNAPSHOT_DATA_INTEGRITY_DISABLED)
+
+
+@pytest.mark.long_test
```
Comment (roger-ryao) [inline manager/integration/tests/test_snapshot.py:252]: Hi @khushboo-rancher :
      I  introduce `long_test` marking for the tests which take more than 10 min
      
Screenshot_20230105_163233

File: manager/integration/tests/test_snapshot.py (hunk 2 above)
Comment (derekbit) [reply manager/integration/tests/test_snapshot.py:287]: Many thanks!

File: manager/integration/tests/test_snapshot.py (hunk 3 above)
Comment (roger-ryao) [reply manager/integration/tests/test_snapshot.py:252]: @derekbit 
It's due to another ticket.[#5199](https://github.com/longhorn/longhorn/issues/5199#issuecomment-1371449299)
We need to introduce some additional marking for our test.

**Comment by khushboo-rancher** [comment]
@roger-ryao Thanks, the marking looks good. However, I was thinking should we have the mark as `long_running` instead `long_test`, that will make more sense.

File: manager/integration/tests/test_snapshot.py (hunk 3 above)
Comment (derekbit) [reply manager/integration/tests/test_snapshot.py:252]: Got it. Thank you.

**Review by derekbit — DISMISSED** [review]

**Comment by roger-ryao** [comment]
> @roger-ryao Thanks, the marking looks good. However, I was thinking should we have the mark as `long_running` instead `long_test`, that will make more sense.
@khushboo-rancher 
Ok, I changed the marking to `long_running`

**Review by c3y1huang — COMMENTED** [review]
Nit: suggest having smaller commits based on the purpose. Ex:
- Add test cases for volume set the snapshot data integrity
- Fix test_snapshot_hash_detect_corruption_in_global_fast_check_mode
- mark long-running tests

**Comment by roger-ryao** [comment]
> Nit: suggest having smaller commits based on the purpose. Ex:
> 
> * Add test cases for volume set the snapshot data integrity
> * Fix test_snapshot_hash_detect_corruption_in_global_fast_check_mode
> * mark long-running tests

Hi @c3y1huang :
    I separate the commit. Please review it again. Thanks

**Review by c3y1huang — APPROVED** [review]

**Comment by roger-ryao** [comment]
@Mergifyio backport v1.4.x

**Pull request merged**